features/dd-sdk-android-flags/src/test/kotlin/com/datadog/android/flags/internal/FlagsStateChannelTest.kt

```diff
@@ -0,0 +1,179 @@
+
+        // When
+        testedChannel.notifyReady()
+
+        // Then
+        verify(mockListener).onStateChanged(FlagsClientState.READY, null)
+    }
+
+    // endregion
+
+    // region notifyReconciling
+
+    @Test
+    fun `M notify listeners with RECONCILING W notifyReconciling()`() {
+        // Given
+        testedChannel.addListener(mockListener)
+
+        // When
+        testedChannel.notifyReconciling()
+
+        // Then
+        verify(mockListener).onStateChanged(FlagsClientState.RECONCILING, null)
+    }
+
+    // endregion
+
+    // region notifyError
+
+    @Test
+    fun `M notify listeners with ERROR and null W notifyError() {no error provided}`() {
+        // Given
+        testedChannel.addListener(mockListener)
+
+        // When
+        testedChannel.notifyError()
+
+        // Then
+        verify(mockListener).onStateChanged(FlagsClientState.ERROR, null)
+    }
+
+    @Test
+    fun `M notify listeners with ERROR and throwable W notifyError() {error provided}`() {
+        // Given
+        testedChannel.addListener(mockListener)
+        val fakeError = RuntimeException("Test error")
+
+        // When
+        testedChannel.notifyError(fakeError)
+
+        // Then
+        verify(mockListener).onStateChanged(FlagsClientState.ERROR, fakeError)
+    }
+
+    // endregion
+
+    // region addListener / removeListener
+
+    @Test
+    fun `M notify listener W addListener() and notify`() {
+        // Given
+        testedChannel.addListener(mockListener)
+
+        // When
+        testedChannel.notifyReady()
+
+        // Then
+        verify(mockListener).onStateChanged(FlagsClientState.READY, null)
+    }
+
+    @Test
+    fun `M not notify listener W removeListener() and notify`() {
+        // Given
+        testedChannel.addListener(mockListener)
+        testedChannel.removeListener(mockListener)
+
+        // When
+        testedChannel.notifyReady()
+
+        // Then
+        verifyNoInteractions(mockListener)
+    }
+
+    @Test
+    fun `M notify all listeners W multiple listeners registered`() {
+        // Given
+        val mockListener2 = org.mockito.Mockito.mock(FlagsStateListener::class.java)
+        testedChannel.addListener(mockListener)
+        testedChannel.addListener(mockListener2)
+
+        // When
+        testedChannel.notifyReady()
+
+        // Then
+        verify(mockListener).onStateChanged(FlagsClientState.READY, null)
+        verify(mockListener2).onStateChanged(FlagsClientState.READY, null)
+    }
+
+    @Test
+    fun `M notify listeners in order W multiple state transitions`() {
+        // Given
+        testedChannel.addListener(mockListener)
+
+        // When
+        testedChannel.notifyNotReady()
+        testedChannel.notifyReconciling()
+        testedChannel.notifyReady()
+
+        // Then
+        val captor = argumentCaptor<FlagsClientState>()
+        verify(mockListener, org.mockito.kotlin.times(3)).onStateChanged(captor.capture(), org.mockito.kotlin.any())
```

Comment:
```suggestion
        verify(mockListener, times(3)).onStateChanged(captor.capture(), any())
```

Although it is better to use `inOrder { verify(...) }` verification, this way no need to use the captor